src/main/connections/mysql/MySqlConnection.ts

```diff
@@ -0,0 +1,50 @@
+import { SqlDefinitionResult, SqlModificationResult, SqlQueryResult } from '@nodescript/adapter-sql-protocol';
+import { FieldPacket, PoolConnection, QueryResult, ResultSetHeader, RowDataPacket } from 'mysql2/promise';
+
+import { SqlError } from '../../global/SqlError.js';
+import { getMySqlTypeByCode } from './MySqlFieldTypeMap.js';
+
+export class MySqlConnection {
+    constructor(
+        protected client: PoolConnection
+    ) {}
+    async define(text: string): Promise<SqlDefinitionResult> {
+        await this.execute(text);
+        return { command: text.split(' ')[0] };
```

Comment:
I think it could be helpful to have a helper method here to extract the SQL command, in case its not in the expected form. For example it may have comments or whitespace in the beginning, or be created with lowercase
E.g. `SELECT * FROM users;` or 
```
    /*
        Multi-line comment
    */
    INSERT INTO orders (id, item) VALUES (1, 'Book');
```
I think having a method that handles trimming, replacing any single or multi line comments, potentially a regex if needed, and uppercasing the final result could help

Reply:
Yeah I've gone with scrapping the field entirely - it was basically just a confirmation for the user but effectively not throwing an error covers that anyway. I'll double check there aren't other string manipulations that could be collated though 👍